### PR TITLE
cannot deploy contracts with linked libraries using waffle's deployContract

### PR DESCRIPTION
**Pull request by clifton**
added minimal failing example

```
cd packages/hardhat-waffle
yarn test
```

happy to fix this but will need a bit of guidance. thanks!

### CODE REVIEW

File: packages/hardhat-waffle/test/fixture-projects/hardhat-project/test/test-link.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+    await expect(contract.inc(7)).to.emit(contract, 'Increment').withArgs(7);
+  });
+
+  it('should be able to link libraries', async () => {
+    const lib = await deployContract(signer, await artifacts.readArtifact('Lib'));
+    expect(lib.address).to.be.properAddress;
+
+    // no straightforward way to use hardhat-waffle's `link` library from here
+    // with deployContract or readArtifact
+    // link(UsesLibArtifact, 'contract/path', lib.address);
```
Comment (alcuadrado) [inline packages/hardhat-waffle/test/fixture-projects/hardhat-project/test/test-link.js:23]: Hey, thanks for sending this PR, @clifton. I love the idea of a failing test as a report/feature request 😍

Can you link to waffles config about this or explain a little more? I'm not sure I understand this.

Comment (clifton) [reply packages/hardhat-waffle/test/fixture-projects/hardhat-project/test/test-link.js:23]: Sure. External libraries from npm packages work fine with deployContract but if you try to include a library from a relative path, it throws an invalid bytecode error on deployment.

I understand that ether’s getContractFactory will work with the link function here, though it’s not clear in the docs or in hardhat’s API what the idiomatic way of instantiating contracts for testing is. If it’s with getContractFactory, why does the deployContract function in hardhat-waffle exist? Or should all linking happen automatically?

Comment (fvictorio) [reply packages/hardhat-waffle/test/fixture-projects/hardhat-project/test/test-link.js:23]: I believe this is a problem in Waffle. I opened an issue on their repo: https://github.com/EthWorks/Waffle/issues/429

Comment (clifton) [reply packages/hardhat-waffle/test/fixture-projects/hardhat-project/test/test-link.js:23]: Yep, makes total sense. Thank you!

**Comment by alcuadrado** [comment]
Hey @clifton,

We just talked about this and decided to close it until https://github.com/EthWorks/Waffle/issues/429. Once that's fixed, we'll merge this PR.

As I mentioned, sending PRs with failing tests is awesome! Thanks for submitting it.

**Pull request closed without merging**